File: SMeatSocialNetwork/SMeat.MODELS/Entities/Message.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using SMeat.MODELS.DTO;
using SMeat.MODELS.Enums;

namespace SMeat.MODELS.Entities
{
    public class Message : Entity 
    {
        public Message()
        {
            Id = Guid.NewGuid().ToString();
        }

        [Key]
        public override string Id { get; set; }

        public DateTimeOffset? DateTime { get; set; }

        public string Text { get; set; }

        public MessageStatus MessageStatus { get; set; }

        [ForeignKey("User")]
        public string UserId { get; set; }

        public virtual User User { get; set; }

        [ForeignKey("Chat")]
        public string ChatId { get; set; }

        public virtual Chat Chat { get; set; }

        [ForeignKey("GroupChat")]
        public string GroupChatId { get; set; }

        public virtual GroupChat GroupChat { get; set; }
    }
}

File: SMeatSocialNetwork/SMeat.WEB/src/app/_models/message.ts
import { User } from "./user"; 

import { MessageStatus } from "../_enums/message-status";
 
export class Message {
    id: string;
    text: string;
    user: User;
    userId: string;
    status: MessageStatus;    
    dateTime: string;
    chatId: string;
    tempId: string;

    constructor(id?: string, text?: string, userId?: string, user?: User, status?: MessageStatus, dateTime?: string, chatId?: string,  tempId?: string) {
        this.id = id;
        this.tempId = tempId;
        this.text = text;
        this.userId = userId;
        this.user = user;
        this.status = status;        
        this.dateTime = dateTime;
        this.chatId = chatId;
    }
}